rtos/mbed_boot.c

```diff
@@ -574,9 +576,14 @@ void pre_main(void)
     singleton_mutex_attr.cb_mem = &singleton_mutex_obj;
     singleton_mutex_id = osMutexNew(&singleton_mutex_attr);
 
+#if (__IAR_SYSTEMS_ICC__ >= 8)
+    __iar_Initlocks();
```

Comment:
As per the release document, void __iar_clearlocks(void); should be called to destroy the locks. It should be called after atexit and _Close_all. 
In mbed applications, we never do close all we just wait for system to die (in case of issue). Other option was to call it on main thread exit, but than timer thread and background thread will still be running.